build.bash

```diff
@@ -0,0 +1,134 @@
+#!/usr/bin/env bash
+
+nproc_default=4
+
+script_name=$(basename "${0}")
+
+show_help() {
+    cat << EOF
+Usage: ./$script_name [OPTIONS]
+
+Build script wrapper around CMake. Supplied arguments that do not match the
+options below will be passed to CMake when generating the build system.
+
+Options:
+      --clean   Delete build directory before invoking CMake.
+      --mpi     Compile MPI executable.
+      --compiler <compiler>
+                Specify the compiler to use.
+  -j <jobs>     Specify the number of parallel jobs in the compilation. By
+                default this value is set to $nproc_default.
+  -h, --help    Show this screen.
+
+Enabling debug mode:
+
+  The release build is default. To enable debug mode, specify the CMake option
+  -DCMAKE_BUILD_TYPE=Debug when invoking $script_name.
+
+Enabling verbose output from Makefile builds:
+
+  To enable more verbose output from Makefile builds, specify the CMake option
+  -DCMAKE_VERBOSE_MAKEFILE=ON when invoking $script_name.
+
+EOF
+}
+
+cmake_args=(-DCMAKE_BUILD_TYPE=Release)
+
+# Argument parsing adapted and stolen from http://mywiki.wooledge.org/BashFAQ/035#Complex_nonstandard_add-on_utilities
+while [ $# -gt 0 ]; do
+    case $1 in
+        --clean)
+            rm -r build
+            ;;
+        --mpi)
+            mpi=1
+            cmake_args+=(-DCABLE_MPI="ON")
+            ;;
+        --compiler)
+            compiler=$2
+            shift
+            ;;
+        -j)
+            CMAKE_BUILD_PARALLEL_LEVEL=$2
+            shift
+            ;;
+        -h|--help)
+            show_help
+            exit
+            ;;
+        ?*)
+            cmake_args+=("$1")
+            ;;
+    esac
+    shift
+done
+
+if hostname -f | grep gadi.nci.org.au > /dev/null; then
+    : "${compiler:=intel-llvm}"
+
+    . /etc/bashrc
+    module purge
+    module add cmake/3.24.2
+    module add netcdf/4.9.2
+    case ${compiler} in
+        intel)
+            module add intel-compiler/2019.5.281
+            compiler_lib_install_dir=Intel
+            [[ -n $mpi ]] && module add intel-mpi/2019.5.281
+            ;;
+        intel-llvm)
+            module add intel-compiler-llvm/2023.0.0
+            compiler_lib_install_dir=Intel
+            [[ -n $mpi ]] && module add intel-mpi/2019.5.281
+
+            # This is required to use ifort instead of ifx (default):
+            cmake_args+=(-DCMAKE_Fortran_COMPILER="ifort")
+
+            # This is required to build with runtime dispatch for all
+            # architectures on Gadi (see
+            # https://opus.nci.org.au/display/Help/Sapphire+Rapids+Compute+Nodes
+            # for more information):
+            cmake_args+=(-DCMAKE_Fortran_FLAGS_RELEASE_INIT="-march=broadwell -axSKYLAKE-AVX512,CASCADELAKE,SAPPHIRERAPIDS -diag-disable=15009")
+            ;;
```

Comment:
Why is this not set also for the other Intel compiler?

Reply:
The [NCI documentation](https://opus.nci.org.au/display/Help/Sapphire+Rapids+Compute+Nodes) states to use these flags with the `intel-compiler-llvm` module. The intel classic compiler used in the `intel-compiler/2019.5.281` module does not support the `-ax*` optimisation flags.